commit cfc404843b7ae1b5c3638bf3ac3ed128bc398bdd
Author: Vitali Lovich <vitali.lovich@palm.com>
Date:   Tue Nov 3 13:47:55 2009 -0800

    Fix-up document generation logic to be more git submodule friendly.

diff --git a/YAJLDoc.cmake b/YAJLDoc.cmake
index 049cdef..c8a1923 100644
--- a/YAJLDoc.cmake
+++ b/YAJLDoc.cmake
@@ -1,6 +1,6 @@
-FIND_PROGRAM(doxygenPath doxygen)
+FIND_PACKAGE(Doxygen)
 
-IF (doxygenPath)
+IF (DOXYGEN_FOUND)
   SET (YAJL_VERSION ${YAJL_MAJOR}.${YAJL_MINOR}.${YAJL_MICRO})
   SET(yajlDirName yajl-${YAJL_VERSION})
   SET(docPath
@@ -13,14 +13,20 @@ IF (doxygenPath)
 
   FILE(MAKE_DIRECTORY "${docPath}")
 
-  ADD_CUSTOM_TARGET(doc
-                    ${doxygenPath} YAJL.dxy   
+  ADD_CUSTOM_TARGET(yajl-doc
+                    ${DOXYGEN_EXECUTABLE} YAJL.dxy   
                     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
 
-ELSE (doxygenPath)
-  MESSAGE("!! doxygen not found, not generating documentation")     
+ELSE (DOXYGEN_FOUND)
+  MESSAGE("!! doxygen not found, not generating documentation")
+   
   ADD_CUSTOM_TARGET(
-    doc
+    yajl-doc
     echo doxygen not installed, not generating documentation
   )
-ENDIF (doxygenPath)
+ENDIF (DOXYGEN_FOUND)
+
+IF(NOT TARGET doc)
+  ADD_CUSTOM_TARGET(doc)
+ENDIF(NOT TARGET doc)
+ADD_DEPENDENCIES(doc yajl-doc)
diff --git a/src/YAJL.dxy b/src/YAJL.dxy
index 680eb43..c848d7d 100644
--- a/src/YAJL.dxy
+++ b/src/YAJL.dxy
@@ -463,7 +463,7 @@ WARN_LOGFILE           =
 # directories like "/usr/src/myproject". Separate the files or directories 
 # with spaces.
 
-INPUT                  =  ../src/yajl ../src/api
+INPUT                  =  @CMAKE_CURRENT_SOURCE_DIR@/src/yajl @CMAKE_CURRENT_SOURCE_DIR@/src/api
 
 # This tag can be used to specify the character encoding of the source files that 
 # doxygen parses. Internally doxygen uses the UTF-8 encoding, which is also the default 
